apps/server/src/shared/testing/factory/external-class-dto.factory.ts

```diff
@@ -0,0 +1,11 @@
+import { ExternalClassDto } from '@modules/provisioning';
+import { ObjectId } from '@mikro-orm/mongodb';
+import { Factory } from 'fishery';
+
+export const externalClassDtoFactory = Factory.define<ExternalClassDto, ExternalClassDto>(
+	({ sequence }) =>
+		new ExternalClassDto({

```

Comment:
Why don't you just do a 'return' here?

Since 'Factory.define<ExternalClassDto..' already specifies the type, explicitly instantiating with 'new ExternalClassDto' is redundant. It also adds complexity with an additional level of instantiation. It also makes maintainability harder, since changes to the constructor of ExternalClassDto require adjustments to all places where 'new ExternalClassDto' is used. And as far as I could see, the other factories don't use it either.

Reply:
cause it needed like that for OauthDataFactory mapping gives back every part as his type, and with out to do it here the expected will have the part as object not ExternalClassDto. and the test will not work.

Reply:
Okay